core/model/modx/transport/modtransportpackage.class.php

```diff
@@ -362,8 +362,26 @@ public function transferPackage($sourceFile, $targetDir) {
         if (is_dir($targetDir) && is_writable($targetDir)) {
             if (!is_array($this->xpdo->version)) { $this->xpdo->getVersionData(); }
             $productVersion = $this->xpdo->version['code_name'].'-'.$this->xpdo->version['full_version'];
-
-            $source= $this->get('service_url') . $sourceFile.(strpos($sourceFile,'?') !== false ? '&' : '?').'revolution_version='.$productVersion;
+            
+            /* make sure the package is downloaded, if not attempt re-download */
+            if (!file_exists($targetDir . $sourceFile)) {
+                /* not very pretty way of getting the download URL, but otherwise we have to make a 
+                 * remote request to $this->Provider->info($this->signature) to get the URL.
+                 * loop trough metadata arrays until location is found */
+                $locationarr = array_filter($this->get('metadata'), function($item) { 
```

Comment:
To maintain readability I suggest using the camelCase naming convention for variable names.